framework/src/utils/MooseEnumBase.C

```diff
@@ -127,3 +107,43 @@ MooseEnumBase::checkDeprecatedBase(const std::string & name_upper) const
       mooseWarning(name_upper + " is deprecated");
   }
 }
+
+std::vector<std::string>
+MooseEnumBase::getNames() const
+{
+  std::vector<std::string> out;
+  out.reserve(_items.size());
+  for (auto it = _items.begin(); it != _items.end(); ++it)
+    out.push_back(it->name());
+  return out;
+}
+
+std::string
+MooseEnumBase::getRawNames() const
+{
+  std::ostringstream oss;
```

Comment:
@dschwen has me converted to his new "stringify" method. You might try that instead.